lms/static/scripts/frontend_apps/components/dashboard/test/DashboardActivityFilters-test.js

```diff
@@ -139,6 +139,11 @@ describe('DashboardActivityFilters', () => {
         usePaginatedAPIFetch: fakeUsePaginatedAPIFetch,
       },
     });
+    // Do not mock ContentWithBadge, as it helps test scenarios where it is
+    // used vs scenarios where it isn't
+    $imports.$restore({
+      './ContentWithBadge': true,
+    });
```

Comment:
Not mocking this helped me catch a bug in the segments filter, where we were displaying a badge with value `0` for assignments with no segments.

These changes addressed that bug https://github.com/hypothesis/lms/pull/6820/files#diff-7f55aaf9066c358488166a2a3393b1346e0f8848e672f4fa1f19e23f08d28977R153-R158

Reply:
To clarify, the purpose of avoiding mocking is so that the badge count appears in the rendered text?